### PR TITLE
LOG-1284: Generate Dockerfile from template

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* adds a script for Dockerfile generation from the midstream Dockerfile.in
* Adds a linter task to be run by CI to check Dockerfile changes

### Links
https://issues.redhat.com/browse/LOG-1284

cc @vimalk78

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,36 @@
...
+ENV BUILD_VERSION=${CI_CONTAINER_VERSION}
+ENV OS_GIT_MAJOR=${CI_X_VERSION}
+ENV OS_GIT_MINOR=${CI_Y_VERSION}
+ENV OS_GIT_PATCH=${CI_Z_VERSION}
+ENV SOURCE_GIT_COMMIT=${CI_ELASTICSEARCH_PROXY_UPSTREAM_COMMIT}
+ENV SOURCE_GIT_URL=${CI_ELASTICSEARCH_PROXY_UPSTREAM_URL}
+ENV REMOTE_SOURCE=${REMOTE_SOURCE:-.}
+
+
+LABEL \
```
Comment (jcantrill) [inline Dockerfile:14]: @vimalk78  I moved this here to make local building better, but will it result in the final image not having these labels?  cc @sosiouxme  do you know?

Comment (vimalk78) [reply Dockerfile:14]: If the `LABEL` is here, i get this error in OSBS
```
FAILED: BuildError: Required LABELs haven't been found in Dockerfile: com.redhat.component (or BZComponent), name (or Name), version (or Version).
```

It builds fine if the `LABEL` is moved to end of file

Comment (jcantrill) [reply Dockerfile:14]: ill check it.  i think the labels have to probably be in the final image

**Pull request merged**